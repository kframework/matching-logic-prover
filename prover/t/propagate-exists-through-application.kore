symbol foo

axiom a: foo(foo(1, \exists{X{Int}} X{Int}), \exists{X{Int}} foo(X{Int}, 3))

claim foo(foo(1, \exists{X{Int}} X{Int}), foo(\exists{X{Int}} X{Int}, 3))
strategy propagate-exists-through-application 1
       . apply(a, fail)
