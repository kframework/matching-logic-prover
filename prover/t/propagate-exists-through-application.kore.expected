<prover>
  <exit-code>
    0
  </exit-code>
  <goals>
    <goal>
      id: ., parent: .
      <claim>
        \and ( .Patterns )
      </claim>
      <k>
        .
      </k>n<expected>
        .
      </expected>
      <local-context>
        .LocalDeclCellSet
      </local-context>
    <trace>
      .
    </trace>
  </goals>
  <declarations>
    <declaration>
      axiom a : foo ( foo ( 1 , \exists { X { Int } , .Patterns } X { Int } , .Patterns ) , \exists { X { Int } , .Patterns } foo ( X { Int } , 3 , .Patterns ) , .Patterns )
    </declaration> <declaration>
      axiom ax0 : foo ( foo ( 1 , \exists { X { Int } , .Patterns } X { Int } , .Patterns ) , foo ( \exists { X { Int } , .Patterns } X { Int } , 3 , .Patterns ) , .Patterns )
    </declaration> <declaration>
      symbol foo ( Int , Int , .Sorts ) : Int
    </declaration>
  </declarations>
</prover>
